src/common/lib/transport/websockettransport.ts

```diff
@@ -39,12 +39,27 @@ class WebSocketTransport extends Transport {
     callback: TryConnectCallback
   ) {
     const transport = new WebSocketTransport(connectionManager, auth, params);
+
+    let transportAttemptTimer: NodeJS.Timeout | number;
     const errorCb = function (this: { event: string }, err: ErrorInfo) {
+      clearTimeout(transportAttemptTimer);
       callback({ event: this.event, error: err });
     };
+
+    const realtimeRequestTimeout = connectionManager.options.timeouts.realtimeRequestTimeout;
```

Comment:
I would have thought you could remove a lot of the duplication by adding a couple of methods to the base transport class (`startTryConnectTimer` and `clearTryConnectTimer` or whatever).

Reply:
istm a halfway-house of sharing just the timer helper functions would be a bit messy -- you'd have to pass in errorCb, the transport instance, and the timeout, the list of event names to deregister varies a bit, etc. If we want to dedup, better would be to go the whole hog and deduplicate the whole of tryConnect -- I'll have a quick go at that

Reply:
https://github.com/ably/ably-js/pull/1035/commits/92e4671c3871c9f43f23784d43367e0f12ddf2fa, https://github.com/ably/ably-js/pull/1035/commits/b2aeea8fe997777494342ffbaf99dcde8fb52ffb -- that went smoother than expected, turns out they were pretty much all identical

Reply:
nice, that's a big improvement thanks 👍